deeplake/core/vectorstore/deeplake_vectorstore.py

```diff
@@ -0,0 +1,213 @@
+        self.dataset = utils.create_or_load_dataset(
+            dataset_path, token, creds, logger, read_only, **kwargs
+        )
+        self._embedding_function = embedding_function
+        self.exec_option = exec_option
+
+    def add(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        embeddings: Optional[Any] = None,
+    ) -> List[str]:
+        """Adding elements to deeplake vector store
+
+        Args:
+            texts (Iterable[str]): texts to add to deeplake vector store
+            metadatas (Optional[List[dict]], optional): List of metadatas.. Defaults to None.
+            ids (Optional[List[str]], optional): List of document IDs. Defaults to None.
+            embeddings (Optional[Any], optional): embedding of texts. Defaults to None.
+        Returns:
+            ids (List[str]): List of document IDs
+        """
+        elements = self._create_elements(ids, texts, metadatas, embeddings)
+        self._run_data_injestion(elements)
+        self.dataset.commit(allow_empty=True)
+        self.dataset.summary()
+        return ids
+
+    def query(
+        self,
+        query: Optional[str] = None,
+        embedding: Optional[float] = None,
+        k: int = 4,
+        distance_metric: str = "L2",
+        filter: Optional[Any] = None,
+        exec_option: Optional[str] = None,
+    ):
+        view = self._attribute_based_filtering(filter)
+        if len(view) == 0:
+            return view
+
+        if self._embedding_function is None and embedding is None:
+            view, scores, indices = self._exact_text_search(view, query)
+        else:
+            emb = embedding or self._embedding_function.embed_query(
+                query
+            )  # type: ignore
+            query_emb = np.array(emb, dtype=np.float32)
+            embeddings = view.embedding.numpy(fetch_chunks=True)
+            indices, scores = vector_search.search(
+                query_embedding=query_emb,
+                embeddings=embeddings,
+                k=k,
+                distance_metric=distance_metric.lower(),
+                exec_option=exec_option or self._exec_option,
+                deeplake_dataset=self.dataset,
+            )
+            return (view, indices, scores)
+
+    def delete(
+        self,
+        ids: Optional[List[str]] = None,
+        filter: Optional[Dict[str, str]] = None,
+        delete_all: Optional[bool] = None,
+    ) -> bool:
+        """Delete the entities in the dataset
+        Args:
+            ids (Optional[List[str]]): The document_ids to delete.
+                Defaults to None.
+            filter (Optional[Dict[str, str]]): The filter to delete by.
+                Defaults to None.
+            delete_all (Optional[bool]): Whether to drop the dataset.
+                Defaults to None.
+        """
+        if delete_all:
+            self.dataset.delete(large_ok=True)
```

Comment:
This should probably be a different method called `drop`

Reply:
Also makes sense to have a `clear` method which deletes all samples without dropping the dataset.

Reply:
I think delete is a good name for this method